src/app/branded/step-1/branded-checkout-step-1.component.js

```diff
@@ -154,12 +161,176 @@ class BrandedCheckoutStep1Controller {
   checkSuccessfulSubmission () {
     if (every(this.submission, 'completed')) {
       if (every(this.submission, { error: false })) {
-        this.next()
+        if (this.useV3 === 'true') {
+          this.submitOrderInternal()
+        } else {
+          this.next()
+        }
       } else {
         this.submitted = false
       }
     }
   }
+
+  loadCart () {
+    this.errorLoadingCart = false
+
+    // Return the observable instead of subscribing here
+    return this.cartService.get().pipe(
+      tap(data => {
+        // Setting cart data and analytics
+        this.cartData = data
+        this.brandedAnalyticsFactory.saveCoverFees(this.orderService.retrieveCoverFeeDecision())
+        this.brandedAnalyticsFactory.saveItem(this.cartData.items[0])
+        this.brandedAnalyticsFactory.addPaymentInfo()
+        this.brandedAnalyticsFactory.reviewOrder()
+        console.log('done loading cart')
+      }),
+      catchError(error => {
+        // Handle errors by setting flag and logging the error
+        this.errorLoadingCart = true
+        this.$log.error('Error loading cart data for branded checkout step 2', error)
+        return Observable.throw(error) // Rethrow the error so the observable chain can handle it
+      })
+    )
+  }
+
+  loadCurrentPayment () {
+    this.loadingCurrentPayment = true
+
+    return this.orderService.getCurrentPayment().pipe(
+      tap(data => {
+        if (!data) {
+          this.$log.error('Error loading current payment info: current payment doesn\'t seem to exist')
+        } else if (data['account-type']) {
+          this.bankAccountPaymentDetails = data
+        } else if (data['card-type']) {
+          this.creditCardPaymentDetails = data
+        } else {
+          this.$log.error('Error loading current payment info: current payment type is unknown')
+        }
+        this.loadingCurrentPayment = false
+      }),
+      catchError(error => {
+        this.loadingCurrentPayment = false
+        this.$log.error('Error loading current payment info', error)
+        return Observable.throw(error) // Propagate error
+      })
+    )
+  }
+
+  checkErrors () {
+    // Then check for errors on the API
+    return this.orderService.checkErrors().pipe(
+      tap((data) => {
+        this.needinfoErrors = data
+        console.log('checking errors')
+      }),
+      catchError(error => {
+        this.$log.error('Error loading checkErrors', error)
+        return Observable.throw(error)
+      }))
+  }
+
+  submitOrderInternal () {
```

Comment:
I'm going to dig deeper, but I'm hoping I can find a way to not have to recreate this logic here by including step-3 on this step-1 page.

Reply:
I'm sure it's possible, but there will be more branching in `step-3`. The tricky part will be changing `step-3` to accept a new prop that lets us save the 3 forms and wait for them to finish saving before doing the submission.

Reply:
Or we could possibly do the handling of the 3 forms in an intermediary, invisible-to-the-user step?

Reply:
I guess, step-3 itself has review data in the template, so we probably don't want to use it directly; however, it might make sense to refactor the common logic out to a service and use it in both places (step-3 and branded-step-1)